lib/Backend/LinearScan.cpp

```diff
@@ -5135,8 +5135,10 @@ void LinearScan::GeneratorBailIn::BuildBailInSymbolList(
 
         if (unrestorableSymbols.TestAndClear(value->m_id))
         {
-            if (this->NeedsReloadingSymWhenBailingIn(copyPropSym.Key()))
+            if (this->NeedsReloadingSymWhenBailingIn(copyPropSym.Value()))
             {
+                // we're restoring this sym based on the value also remove the key from unrestorableSymbols
+                unrestorableSymbols.Clear(key->m_id);
```

Comment:
We're restoring from the Value, get rid of the key as well (otherwise hit a book keeping failfast)